src/App.tsx

```diff
@@ -17,8 +17,11 @@ const STEP_MS = 100;
 
 @observer
 class App extends React.Component<AppProps> {
+  currentTime: number;
```

Comment:
It looks like this change adds time tracking to `App` and `Substance`. Since it's a global value needed in a handful of places, I think it makes sense to just track in the `RootStore`. That's what my change does, along with passing the current time in `RootStore` down to the `step` functions.